### PR TITLE
Code Review

### PR DESCRIPTION
**Pull request by Chaoskil**
lets see if this works...

### CODE REVIEW

File: Assets/Scripts/Checkpoint.cs [hunk 1]
```diff
@@ -0,0 +1,62 @@
+﻿using System.Collections;
+using System.Collections.Generic;
+using UnityEngine;
+
+public class Checkpoint : MonoBehaviour {
+
+
+    //variable that needs to be applied within unity
+    [SerializeField]
+    private float inactivatedScale = 1, activatedScale = 1.5f;
```
Comment (TheRealNigl) [inline Assets/Scripts/Checkpoint.cs]: Either assign the variables in the Start Function or serialize the field. I recommend not serializing a field.

File: Assets/Scripts/Checkpoint.cs [hunk 2]
```diff
@@ -0,0 +1,62 @@
...
+
+
+    //other variables that dont need to be applied in Unity
+    private bool isActivated = false;
```
Comment (TheRealNigl) [inline Assets/Scripts/Checkpoint.cs]: Assign variables in the Start function

File: Assets/Scripts/Checkpoint.cs [hunk 3]
```diff
@@ -0,0 +1,62 @@
...
+
+        transform.localScale = Vector3.one * scale;
+    }
+
+    //checks if the player touches the checkpoint then makes it the current checkpoint they will respawn at
+    private void OnTriggerEnter2D(Collider2D collision)
+    {
+        if (collision.CompareTag("Player") && !isActivated)
+        {
+            RobotControllerScript player = collision.GetComponent<RobotControllerScript>();
```
Comment (TheRealNigl) [inline Assets/Scripts/Checkpoint.cs]: A simpler statement is "collision.GetComponent<RobotControllerScript>().SetCurrentCheckpoint(this);"

File: Assets/Scripts/Collectibles.cs [hunk 4]
```diff
@@ -0,0 +1,45 @@
+﻿using System.Collections;
+using System.Collections.Generic;
+using UnityEngine;
+
+public class Collectibles : MonoBehaviour {
+
+    //variable that needs to be set inside of Unity
+    [SerializeField]
+    private Animator anim;
```
Comment (TheRealNigl) [inline Assets/Scripts/Collectibles.cs]: GetComponent<Animator>(); in Start?

File: Assets/Scripts/Collectibles.cs [hunk 5]
```diff
@@ -0,0 +1,45 @@
...
+    //once the player touchers the collectible then the collectible is destroyed and the player score is increased
+    private void OnTriggerEnter2D (Collider2D collision)
+    {
+        if (collision.gameObject.CompareTag("Player"))
+        {
+            //plays the animation for the collectible, turns off its hitbox, plays its sound, and increases player score
+            anim.SetBool("Collected", true);
+            boxCollider2D.enabled = false;
+            coinCount++;
+            Debug.Log("Coin count: " + coinCount);
```
Comment (TheRealNigl) [inline Assets/Scripts/Collectibles.cs]: Please remove Debug.log

File: Assets/Scripts/Door.cs [hunk 6]
```diff
@@ -0,0 +1,66 @@
...
+using UnityEngine.SceneManagement;
+
+public class Door : MonoBehaviour {
+
+    //variables that need to be set within Unity
+    [SerializeField]
+    private string sceneToLoad;
+
+    [SerializeField]
+    private Animator anim;
```
Comment (TheRealNigl) [inline Assets/Scripts/Door.cs]: GetComponent<Animator>(); in Start?

File: Assets/Scripts/Door.cs [hunk 7]
```diff
@@ -0,0 +1,66 @@
...
+
+    //variables that do not need to be set within Unity
+    private bool isPlayerInTrigger;
+    private bool openDoor = false;
```
Comment (TheRealNigl) [inline Assets/Scripts/Door.cs]: Initialization in Start?

File: Assets/Scripts/Keys.cs [hunk 8]
```diff
@@ -0,0 +1,48 @@
...
+    { 
+        if (collision.gameObject.CompareTag("Player"))
+        {
+            //gets access to the players script and tells it that the player has the key
+            RobotControllerScript player = collision.GetComponent<RobotControllerScript>();
+            player.DoesPlayerHaveKey(true);
+
+            //turns off the keys hitbox and sends a message saying that the player has the key, plays the key sound as well
+            boxCollider2D.enabled = false;
+            Debug.Log("You have the key!");
```
Comment (TheRealNigl) [inline Assets/Scripts/Keys.cs]: Please remove debug.log

File: Assets/Scripts/Keys.cs [hunk 9]
```diff
@@ -0,0 +1,48 @@
...
+    }
+
+    //Once the player touches the key then they collect it similar to the collectibles
+    private void OnTriggerEnter2D(Collider2D collision)
+    { 
+        if (collision.gameObject.CompareTag("Player"))
+        {
+            //gets access to the players script and tells it that the player has the key
+            RobotControllerScript player = collision.GetComponent<RobotControllerScript>();
+            player.DoesPlayerHaveKey(true);
```
Comment (TheRealNigl) [inline Assets/Scripts/Keys.cs]: collision.GetComponent<RobotControllerScript>().DoesPlayerHaveKey(true);

File: Assets/Scripts/RobotControllerScript.cs [hunk 10]
```diff
@@ -0,0 +1,204 @@
...
+    private float move;
+    private bool facingRight = true;
+    private bool grounded = false;
+    private bool died = false;
+    private float groundRadius = 0.2f;
+    private Checkpoint currentCheckpoint;
+    private int beforeRespawn = 5000;
+    private int timer = 0;
+    private AudioSource audioSource;
+    private bool pickUpKey = false;
```
Comment (TheRealNigl) [inline Assets/Scripts/RobotControllerScript.cs]: Initializations in Start!!!

File: Assets/Scripts/RobotControllerScript.cs [hunk 11]
```diff
@@ -0,0 +1,204 @@
...
+    private Transform groundCheck;
+
+    [SerializeField]
+    private LayerMask whatIsGround;
+
+    [SerializeField]
+    private PhysicsMaterial2D playerMovingPhysicsMaterial, playerStoppingPhysicsMaterial;
+
+    [SerializeField]
+    private Collider2D playerGroundCollider;
```
Comment (TheRealNigl) [inline Assets/Scripts/RobotControllerScript.cs:21]: Consider getting the above Unity components with GetComponent<*SpecificComponent*>()

**Review by TheRealNigl — COMMENTED** [review]
Please take a look at the comments. Let me know your thoughts

File: Assets/Scripts/Checkpoint.cs (hunk 3 above)
Comment (Chaoskil) [reply Assets/Scripts/Checkpoint.cs]: except it doesn't work when I replace it with what your suggesting...

File: Assets/Scripts/Collectibles.cs (hunk 4 above)
Comment (Chaoskil) [reply Assets/Scripts/Collectibles.cs]: "GetComponent<>();" please specify the exact way to use it.

File: Assets/Scripts/Checkpoint.cs (hunk 3 above)
Comment (Chaoskil) [reply Assets/Scripts/Checkpoint.cs]: needed the <RobotControllerScript> with it.

File: Assets/Scripts/RobotControllerScript.cs (hunk 11 above)
Comment (Chaoskil) [reply Assets/Scripts/RobotControllerScript.cs:21]: There's the way to write it, thank you.

Comment (Chaoskil) [reply Assets/Scripts/RobotControllerScript.cs:21]: changed as many as I could about 4 - 5 of them i think.

**Comment by Chaoskil** [comment]
fixed everything (i think) that was commented on.

**Comment by TheRealNigl** [comment]
Sorry for not saying to accept. I just saw the email for the merged branch. You have good coding practices, no overused variables, and a good focus on functions for class. I would recommend to go for more private variables, then use resources.load, get file text, vectors, or floats assignment for the variable initialization. Having a fluid design between classes is important, which I mostly see, finding your style will help you convey the information.

**Pull request merged**